test/phoenix_ecto/plugs/check_repo_ready_test.exs

```diff
@@ -0,0 +1,214 @@
+defmodule Phoenix.Ecto.Plugs.CheckRepoReadyTest do
+  use ExUnit.Case, async: true
+  use Plug.Test
+
+  alias Phoenix.Ecto.Plugs.CheckRepoReady
+
+  defmodule LongLivedProcess do
+    def run do
+      Process.sleep(1_000)
+      run()
+    end
+  end
+
+  defmodule StorageUpRepo do
+    defmodule Adapter, do: def(storage_status(_repo_config), do: :up)
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  defmodule StorageDownRepo do
+    defmodule Adapter, do: def(storage_status(_repo_config), do: :down)
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  defmodule NoStorageStatusRepo do
+    defmodule Adapter, do: nil
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  defmodule StorageErrorRepo do
+    defmodule Adapter,
+      do: def(storage_status(_repo_config), do: {:error, %{message: "Adapter error"}})
+
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  test "does not raise an error when the storage is created and there are no pending migrations" do
+    Process.register(self(), StorageUpRepo)
+    Application.put_env(:check_repo_ready, :ecto_repos, [StorageUpRepo])
```

Comment:
I found this way to fake the `repo` as an alive process since we [check it here](https://github.com/phoenixframework/phoenix_ecto/pull/122/files#diff-1dae4d10e995f5c2fe4a70e09dfd9c61R26) but I'm not really digging this approach, maybe you folks have better ideas? 👀

Reply:
I think this is good! Maybe we should just encapsulate it in a function and use `on_exit` for the `delete_env` clean up.

Reply:
As we discussed in slack, because of this https://github.com/phoenixframework/phoenix_ecto/pull/122#discussion_r344195343 we can't use the `on_exit` since it runs on another process and we can't unregister other processes.

So I will leave it as is for now, cool? 😄

Reply:
Cool!